### PR TITLE
fix!: introduce v4 provider migration code

### PR DESCRIPTION
**Pull request by MSalopek**
Epochs code changes the paramaters store, so the new parameter needs to be added to the store via a migration.

Had v4.0.0 and v4.1.0 stayed the same consensus version, provider chains using v4.0.0 would panic when migrating to v4.1.0 because of the missing migration.

To mitigate this issue provider consensus version got bumped to `4` and the migration code was refactored to introduce the `v4` migration.

### CODE REVIEW

File: x/ccv/provider/migrations/migrator.go [hunk 1]
```diff
@@ -29,6 +30,12 @@ func (m Migrator) Migrate1to2(ctx sdktypes.Context) error {
 
 // Migrate2to3 migrates x/ccvprovider state from consensus version 2 to 3.
 func (m Migrator) Migrate2to3(ctx sdktypes.Context) error {
-	v3.MigrateParams(ctx, m.paramSpace)
```
Comment (insumity) [inline x/ccv/provider/migrations/migrator.go:32]: Why is this removed?

**Review by sainoe — APPROVED** [review]

**Review by mpoke — APPROVED** [review]

File: x/ccv/provider/migrations/migrator.go (hunk 1 above)
Comment (MSalopek) [reply x/ccv/provider/migrations/migrator.go:32]: Because it was moved to v4 (migration used for ICS @ v4.1).

`Migrate2to3` is for v4.0.0.
`Migrate3to4` is for v4.1.0.
